core/src/main/scala/org/apache/spark/Dependency.scala

```diff
@@ -65,15 +65,17 @@ abstract class NarrowDependency[T](_rdd: RDD[T]) extends Dependency[T] {
  * @param keyOrdering key ordering for RDD's shuffles
  * @param aggregator map/reduce-side aggregator for RDD's shuffle
  * @param mapSideCombine whether to perform partial aggregation (also known as map-side combine)
+ * @param isContinuous mark the dependency is base for continuous processing or not
```

Comment:
If we need a bit to mark whether it's continuous or not, ContinuousShuffleDependency shouldn't be inheriting ShuffleDependency in the first place.

Reply:
Actually we can implement ContinuousShuffleDependency by inheriting Dependency. In current way few interface can be changed.

Reply:
Considering ContinuousShuffleDependency to be a special kind of ShuffleDependency doesn't make sense if we need to add this much special casing to do it.

Reply:
Got it, if we think currently implementation is tricky here, we can change the implementation by getting rid of the ContinuousShuffleDependency, just as you said in jira comments :"We might not need this to be an actual org.apache.spark.Dependency."